### ISSUE TITLE
Sloow docker exec and slow access to bind mount 

### ISSUE THREAD

**Issue by aguedeney** [docker/for-mac#4963]
- [x ] I have tried with the latest version of my channel (Stable or Edge)
  - [ x] I have uploaded Diagnostics
  - Diagnostics ID:5242091C-6F73-4036-9B22-4C5956610FAA/20201002044325

### Expected behavior
Take about 5 seconds or more to do a "docker exec"
### Actual behavior
About a 1 second
### Information

  - macOS Version:

### Diagnostic logs

```
Docker for Mac: version...
Catalina 10.15.7
```

### Steps to reproduce the behavior

  1. Launch any container that has a bash shell
  2. The delay (5+ seconds) occurs

**Comment by aguedeney** [comment]
I did a "Clean / Purge data" and things started to work as expected (for now)

**Comment by stephen-turner** [comment]
Maybe a dupe of #4957?

**Comment by aguedeney** [comment]
Looks like it is ... good to know I’m not the only one. Thanks

**Comment by djs55** [comment]
@aguedeney thanks again for your report. Has everything continued working as expected ever since you did a "Clean / Purge data"? If not, could you reproduce the issue (e.g. run the slow commands a few times) and then upload new diagnostics?

Do you happen to have a repro example which you could share with me?

Thanks for all your help so far!

**Comment by aguedeney** [comment]
@djs55 Thanks for your interest in this issue. I'm using the INOTIFY Java API (NIO newWatchService) extensively in a container image.  Version (stable) 2.4.0.0 caused high cpu utilization (230%+ constant) when I ran the service.  I reinstalled Version (stable) 2.3.0.5 and the issue went away. I believe INOTIFY maybe an issue (specultation) with new changes.  What is also interesting is that the same high cpu utilization occurred with gRPC enable or disabled.  At the moment, my workaround is to stick to 2.3.0.5.  Again thanks for the interest.  I'm glad to help out if I can.
